src/autopas/containers/verletListsCellBased/verletListsCells/PairwiseVerletNeighborList.h

```diff
@@ -0,0 +1,140 @@
+   * */
+  using listType = typename VerletListsCellsHelpers<Particle>::PairwiseNeighborListsType;
+  /**
+   * Constructor for PairwiseVerletNeighborList. Initializes private attributes.
+   * */
+  PairwiseVerletNeighborList() : _aosNeighborList{}, _particleToCellMap{}, _globalToLocalIndex{} {}
+
+  /**
+   * @copydoc VerletListsCellsNeighborListInterface::getContainerType()
+   * */
+  [[nodiscard]] ContainerOption getContainerType() const override { return ContainerOption::pairwiseVerletLists; }
+
+  /**
+   * @copydoc VerletListsCellsNeighborListInterface::getNumberOfPartners()
+   * */
+  const size_t getNumberOfPartners(const Particle *particle) const override {
+    size_t localSize = 0;
+    const auto [firstCellIndex, particleInCellIndex] = _particleToCellMap.at(const_cast<Particle *>(particle));
+    size_t numberOfCellsToInteract = 27;
+    for (size_t secondCellIndex = 0; secondCellIndex < numberOfCellsToInteract; secondCellIndex++) {
+      localSize += _aosNeighborList[firstCellIndex][secondCellIndex][particleInCellIndex].second.size();
+    }
+    return localSize;
+  }
+
+  /**
+   * Returns the neighbor list in AoS layout.
+   * @return Neighbor list in AoS layout.
+   * */
+  typename VerletListsCellsHelpers<Particle>::PairwiseNeighborListsType &getAoSNeighborList() {
+    return _aosNeighborList;
+  }
+
+  /**
+   * @copydoc VerletListsCellsNeighborListInterface::buildAoSNeighborList()
+   * */
+  void buildAoSNeighborList(LinkedCells<Particle> &linkedCells, bool useNewton3, double cutoff, double skin,
+                            double interactionLength, const TraversalOption buildTraversalOption) override {
+    _aosNeighborList.clear();
+    auto &cells = linkedCells.getCells();
+    auto cellsSize = cells.size();
+    _aosNeighborList.resize(cellsSize);
+    _globalToLocalIndex.resize(cellsSize);
+
+    for (size_t firstCellIndex = 0; firstCellIndex < cellsSize; ++firstCellIndex) {
+      size_t numCellsInteracting = 27;  // every cell has max 26 neighbors + interaction with itself
+      _aosNeighborList[firstCellIndex].resize(numCellsInteracting);
+      for (size_t secondCellIndex = 0; secondCellIndex < numCellsInteracting; ++secondCellIndex) {
+        // reserve vector of neighbor lists for every particle in cell1
+        _aosNeighborList[firstCellIndex][secondCellIndex].reserve(cells[firstCellIndex].numParticles());
+        size_t particleIndexCurrentCell = 0;
+        for (auto particleIter = cells[firstCellIndex].begin(); particleIter.isValid(); ++particleIter) {
+          // for each particle in cell1 make pair of particle and neighbor list
+          Particle *currentParticle = &*particleIter;
+          _aosNeighborList[firstCellIndex][secondCellIndex].emplace_back(
+              std::make_pair(currentParticle, std::vector<Particle *>()));
+
+          // magic number 5 doesn't make sense here anymore
+          // how much should we actually reserve?
```

Comment:
excellent question! I have no Idea :D The five was found empirically. So probably it is for the best just to try out a few values.